### PR TITLE
LibWeb: Harmonize look of range input element

### PR DESCRIPTION
**Pull request by simonkrauter**
Previously the entire slider track was colored.
Now only the lower part of the slider track (left side of the thumb) is colored.
Chrome and Firefox do the same.

Implementation note:
I called the new shadow dom element "-moz-range-progress", because I have not found the corresponding name with the "-webkit" prefix.

Screenshot Ladybird before:
ladybird_before

Screenshot Ladybird after:
ladybird_after

Screenshot Chrome:
chrome

Screenshot Firefox:
firefox

### CODE REVIEW

**Comment by simonkrauter** [comment]
Update: Set the length of the blue bar to 0, when the value is NaN.

**Comment by bplaat** [comment]
Nice work on improving the `input[type=range]`, but it is a bit strange that we no use a `-moz-` pseudo element without supporting all of them. Currently we only support `-webkit-` pseudo elements. We need to think about if we want to support `-moz-` stuff in the future, and what takes priority? For now it seems better to create an internal pseudo element, I think.

File: Userland/Libraries/LibWeb/HTML/HTMLInputElement.h
Comment (bplaat) [inline Userland/Libraries/LibWeb/HTML/HTMLInputElement.h:323]: You also need to visit this GC item in `HTMLInputElement::visit_edges()`

**Review by bplaat — CHANGES_REQUESTED** [review]

**Comment by simonkrauter** [comment]
Update:
- use `-webkit-progress-value` instead of `-moz-range-progress`
- add m_range_progress_element to visit_edges()

**Comment by bplaat** [comment]
Nice work, small nitpic comment: I don't know if `-webkit-progress-value` is a correct fit, maybe `-libweb-slider-progress` is better. But I'll let the maintainers decide on that.

**Comment by simonkrauter** [comment]
Oh, now I saw, that `-webkit-progress-value` is meant for the `<progress>` element :(

**Comment by simonkrauter** [comment]
Update 3:
- Assign the style for the shadow DOM element directly and don't use any pseudo CSS class name

**Review by bplaat — APPROVED** [review]
Haha, looks good to me

**Review by ADKaster — APPROVED** [review]

**Pull request merged**
